### PR TITLE
Model 11 molprobity realspacerefine

### PR DESCRIPTION
**Pull request by mmmtnez**
(Reviewing this pull request, you can ignore model_10_superpose_pdbs)
=================================================================================
Dear all,

here you are this new pull request model_11_molprobity_realspacerefine, that includes the changes already mentioned in model_10_superpose_pdbs pull request (shown below), and some additional ones:

1.-  Two new phenix package-derived protocols, one to validate pdbs (molprobity), and another one to refine pdbs in the real space and validate them by molprobity (real-space refine). MolProbity validation values are showed in the summary panel of SCIPION. Since MolProbity from Phenix package also computes correlations between map and model (if the map is probided), these two protocols share many aspects in the generation of .pkl data files, as well in retrieving data, and in the visualization. Viewer and test for these protocols have been also included. Tests for this part:
scipion test em.packages.phenix.tests.test_protocol_molprobity.TestMolprobityValidation2
scipion test em.packages.phenix.tests.test_protocol_real_space_refine.TestPhenixRSRefine

2.- MolProbity has also been included in the modeling workflow. Test for this part: 
scipion test tests.em.workflows.test_workflow_modeling.TestMolprobityValidation

3.- In order to homogenize visualization forms for validation protocols (emringer and molprobity), the possibility of visualize map and model by chimera has also been included in the phenix emringer protocol. Tests for this part (individual in the package and in the modeling workflow): 
scipion test em.packages.phenix.tests.test_protocol_emringer.TestEMRingerValidation2
scipion test tests.em.workflows.test_workflow_modeling.TestEMRingerValidation

=================================================================================
                                               model_10_superpose_pdbs  included:
=================================================================================

1.- A new phenix package-derived protocol/tool to handle pdbs (protocol_superpose_pdbs). It allows to superpose a couple of atomic structures (pdbs or cifs) (fixed and moving) to ensure the optimal matching. RMSD between these two pdbs at the beginning (start) and at the end (final) of matching assess the similarity between both structures. RMSD values are showed in the summary panel of SCIPION. Viewer and test for this protocol have been also included. Test for this part:
scipion test em.packages.phenix.tests.test_protocol_superpose_pdbs.TestProtSuperposePdbs

2.- Individual tests for programs that had previously tests in the modeling workflow (test_workflow_modeling) in order to have also specific tests for ccp4, chimera and phenix packages considering potential future plugins. Test for this part:
scipion test em.packages.ccp4.tests.test_protocol_coot_refmac.TestCootRefinement2
scipion test em.packages.ccp4.tests.test_protocol_coot_refmac.TestRefmacRefinement2
scipion test em.packages.chimera.tests.test_protocol_chimera_fit.TestChimeraFit2
scipion test em.packages.phenix.tests.test_protocol_emringer.TestEMRingerValidation2

3.- Modifications in some tests included in the modeling workflow to include changes in ccp4 protocols (coot and refmac) showed below. Tests for this part:
scipion test tests.em.workflows.test_workflow_modeling.TestCootRefinement
scipion test tests.em.workflows.test_workflow_modeling.TestRefmacRefinement
scipion test tests.em.workflows.test_workflow_modeling.TestEMRingerValidation

4.- Modifications in the ccp4 package:
a) pyworkflow/em/packages/ccp4/convert.py: Inclusion of valid CCP4 version.
b) pyworkflow/em/packages/ccp4/protocol_coot.py: Inclusion of a database (in the extra folder) to save coot outputs, and inclusion of the option of adding a label associated to each generated pdb that will be saved by coot scipion_write().
c) pyworkflow/em/packages/ccp4/protocol_refmac.py (and respective protocol templates): Inclusion of the option to have the mask of the atomic structure that has to be refined. Modification to get a functional protocol with cif files of atomic structures.
d) coot and refmac viewers have been also modified according to new protocols.

5.- Modifications in the phenix package:
a) pyworkflow/em/packages/phenix/convert.py: Inclusion of an additional path to execute phenix programs. This new path is needed to run superpose_pdbs and real_space_refine phenix programs, whereas emringer and molprobity run with the already existing path. pyworkflow/em/packages/phenix/protocol_emringer.py has been also modified according with this last change.

### CODE REVIEW

File: pyworkflow/em/packages/ccp4/protocol_coot.py [hunk 1]
```diff
@@ -249,9 +285,10 @@ def createOutputStep(self, inVolumes, norVolumesNames, init_counter=1):
                 counter += 1
                 self._defineOutputs(**outputs)
                 self._defineSourceRelation(inVol, outVol)
+
         if os.path.isfile(self._getExtraPath('STOPPROTCOL')):
             self.setStatus(STATUS_FINISHED)
-            # NOTE: (ROB) can a derthy way to make an interactive process finish but I do not
+            # NOTE: (ROB) can a derty way to make an interactive process finish but I do not
```
Comment (delarosatrevin) [inline pyworkflow/em/packages/ccp4/protocol_coot.py]: dirty?

File: pyworkflow/protocol/params.py [hunk 2]
```diff
@@ -131,6 +131,7 @@ def iterParams(self):
             yield (name, self._form.getParam(name))
 
     def addParam(self, paramName, ParamClass, **kwargs):
+        # type: (object, object, object) -> object
```
Comment (delarosatrevin) [inline pyworkflow/protocol/params.py]: ???

File: pyworkflow/protocol/params.py [hunk 3]
```diff
@@ -230,6 +231,7 @@ def registerParam(self, paramName, param):
         self._analizeCondition(paramName, param)
         
     def addParam(self, *args, **kwargs):
+        # type: (object, object) -> object
```
Comment (delarosatrevin) [inline pyworkflow/protocol/params.py]: ???

File: config/templates/protocols.template [hunk 4]
```diff
@@ -1,256 +1,253 @@
-# -*- conf -*-
```
[the remark this answers is not in the document]
Comment (mmmtnez) [reply config/templates/protocols.template]: Done: written line 1

File: pyworkflow/em/packages/ccp4/protocol_coot.py (hunk 1 above)
Comment (mmmtnez) [reply pyworkflow/em/packages/ccp4/protocol_coot.py]: FIXED: you are right should be dirty

File: pyworkflow/protocol/params.py (hunk 2 above)
Comment (mmmtnez) [reply pyworkflow/protocol/params.py]: line removed

File: pyworkflow/protocol/params.py (hunk 3 above)
Comment (mmmtnez) [reply pyworkflow/protocol/params.py]: line removed

**Pull request merged**